cms/admin/utils.py

```diff
@@ -696,6 +699,51 @@ def save_model(self, request: HttpRequest, obj: models.Model, form: forms.Form,
             setattr(form._content_instance, self.grouper_field_name, obj)
             form._content_instance.save()
 
+    def get_search_fields(self, request):
+        """Return search fields for either grouper model or content model"""
+        content_search_fields = []
+        grouper_search_fields = []
+        for field_name in self.search_fields:
+            if field_name.startswith(f"{self.content_related_field}__") or field_name.startswith("content__"):
+                content_search_fields.append(re.sub(r"^([@^=]?)[^_]+__", r"\1", field_name))
+
+            else:
+                grouper_search_fields.append(field_name)
+
+        if getattr(self, "_content_fields", False):
+            return content_search_fields
+
+        return grouper_search_fields
+
+    def get_search_results(self, request, queryset, search_term):
+        # Reset _content_fields flag
+        self._content_fields = False
+        grouper_search_result, may_have_duplicate_grouper = super().get_search_results(request, queryset, search_term)
+
+        # Set _content_fields for get_search_fields to return content model search fields
+        self._content_fields = True
+        search_result_from_content, may_have_duplicate_content = self._get_content_search_result(
+            request, queryset, search_term
+        )
+
+        return grouper_search_result | search_result_from_content, (
+            may_have_duplicate_grouper & may_have_duplicate_content
+        )
+
+    def _get_content_search_result(self, request, queryset, search_term):
+        """Get search results from content model"""
+        content_queryset = self.content_model.admin_manager.all()
+        content_search_fields = self.get_search_fields(request)
+        content_search_result, may_have_duplicate_content = self.content_model.admin_manager.none(), False
```

Comment:
```suggestion
        content_search_result, may_have_duplicate_content = self.content_model.admin_manager.none(), False
```
Can we put this into an else clause?